manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/PodSpecEnvUtil.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.utils;
+
+import io.hyscale.commons.models.DecoratedArrayList;
+import io.hyscale.servicespec.commons.model.service.Props;
+import io.hyscale.servicespec.commons.model.service.Secrets;
+import io.kubernetes.client.models.V1ConfigMapKeySelector;
+import io.kubernetes.client.models.V1EnvVar;
+import io.kubernetes.client.models.V1EnvVarSource;
+import io.kubernetes.client.models.V1SecretKeySelector;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class PodSpecEnvUtil {
+
+    public static List<V1EnvVar> getPropEnv(Props props, String configMapName) {
```

Comment:
ConfigMapDataUtil return List<ManifestSnippet> but PodSpecEnvUtil returns List<V1EnvVar> 
Why is that difference coming in? Also , can you think of making all these utils compromise return type either to <T> or List<ManifestSnippets> and make them generic under a single interface.